src/components/controls/km/km-validation-errors.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <div class="scbd-controls km-validation-errors document-validation">
+    <div v-if="isAnalyzing" class="alert alert-info" role="alert">
+      <i class="bi bi-gear"></i> {{ t("analyzingDocument") }}
+    </div>
+    <div v-if="isSaving" class="alert alert-info" role="alert">
+      <i class="bi bi-gear"></i> {{ t("savingDocument") }}
+    </div>
+    <div
+      v-if="!isAnalyzing && !isSaving && !errors"
+      class="alert alert-success"
+      role="alert"
+    >
+      {{ t("valid") }}
+    </div>
+    <div v-if="!isAnalyzing && !isSaving">
+      <div
+        v-if="!hideErrors && errors && errors.length"
+        class="alert alert-warning"
+        role="alert"
+      >
+        <button
+          type="button"
+          @click="onButtonClose"
+          aria-label="Close"
+          class="position-absolute top-0 end-0 m-2 bg-transparent border-0 fs-5 text-secondary">
+          <span aria-hidden="true">&times;</span>
+        </button>
+        <strong>
+          {{ tc("contain", errors.length) }} {{ errors.length }} {{ tc("error", errors.length) }}
+        </strong>
+
+        <ul>
+          <li v-for="error in errors" :title="error.parameters" :key="error">
+            <span>{{ t(getTranslation(error)) }}</span>
+            <ul v-if="error.properties">
+              <li
+                v-for="property in error.properties"
+                :title="error.parameters"
+                :key="property"
+              >
+                <a
+                  class="text-decoration-none"
+                  href="#"
+                  @click.prevent="jumpTo($event, property.property)"
+                >
+                  {{ getLabel(property.property) }}
+                </a>
+              </li>
+            </ul>
+            <span v-if="!error.properties">
+              <a
+                class="text-decoration-none"
+                href="#"
+                @click.prevent="jumpTo($event, error.property)"
+              >
+                {{ getLabel(error.property) }}
+              </a>
+            </span>
+          </li>
+        </ul>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup>
+import {useAttrs, defineEmits, defineProps, computed, inject, ref} from "vue";
+import { useI18n } from '../../../services/composables/i18n'
+
+const { t, tc, locale } = useI18n();
+
+const props = defineProps({
+  errors: { type: Array, required: true },
+  isSaving: { type: Boolean, required: false },
+  isAnalyzing: { type: Boolean, required: false },
+  allowShowErrors: { type: Boolean, default: true },
+});
+const hideErrors = ref(false);
+const $emits = defineEmits(["onJumpTo"]);
+const showErrors = computed(() =>!hideErrors && props.errors && props.errors.length && props.allowShowErrors);
+const container = useAttrs().container ?? "body,html";
```

Comment:
not used anywhere

Reply:
can you logically group 
1. props
2. emits
3. computed
4. local variables

Reply:
logically grouped the imports and methods